diff --git a/lib/index.js b/lib/index.js
index c79bc0c..4c5ec81 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -122,7 +122,7 @@ function resolveRgPath() {
 		const executable = parse(process.execPath);
 		const executableSidecar = process.platform === "win32" ? join(executable.dir, `${executable.name}-rg.exe`) : `${process.execPath}-rg`;
 		if ("pkg" in process && existsSync(executableSidecar)) return executableSidecar;
-		const dependency = (await import("@vscode/ripgrep")).rgPath;
+		const dependency = process.env.DSH_RG_PATH || (await import("@vscode/ripgrep")).rgPath; // dsh-harmonyos
 		return process.versions.electron === void 0 ? dependency : dependency.replace(/\.asar(?=[\\/])/u, ".asar.unpacked");
 	});
 	return rgPathPromise;
